spec/support/output_helper.rb

```diff
@@ -1,15 +1,27 @@
 # frozen_string_literal: true
 
 module OutputHelper
-  def expect_output(expected)
-    expected.each do |expected_line|
-      line = @runner.readline
-      expect(line).to match(expected_line), @runner.output.to_s
-    end
+  def section(key)
+    @runner.output.section(key)
   end
 
-  def expect_newline
-    expect(@runner.readline).to match(/^\n/), @runner.output.to_s
+  def expect_section(section_key, expected)
+    actual_section = section(section_key)
+    section_lines = actual_section.split("\n")
+    unless section_lines.length == expected.length
+      raise "Not enough expectations given to `expect_section` for section #{section_key}. " \
+        "Number of expectations given: #{expected.length}. Section contents:\n#{actual_section}"
+    end
+
+    section_lines.each_with_index do |actual_line, index|
+      expectation = expected[index]
+      case expectation
+      when Regexp
+        expect(actual_line).to match(expectation), @runner.output.to_s
```

Comment:
I don't know. I don't need to always match everything. I don't need a pixel perfect test setup that checks if all integrations print it exactly the same. It's less important than the JSON it sends to the server in my opinion.

For comparison, see how I wrote [the diagnose tests for the Ruby gem for example](https://github.com/appsignal/appsignal-ruby/blob/cda07141cb64036e09ba3fb29cbdd5acf4d42bd3/spec/lib/appsignal/cli/diagnose_spec.rb). I test the full format in some cases, but in most tests I just check if a string occurs anywhere in the output.

I think we can add the additional matchers `^` and `$` where relevant, but I don't want to always match every line exactly. This comes mostly from my idea of testing separate things in matchers rather than full lines all the time.

For example, given the config option line `active: true (source: initial)` (with 4 leading spaces). I'd write a test to test the format (with the leading spaces) and in another test I'd have an assertion that only test the different values as they are printed. I don't need to test every time that `active: true` has 4 leading spaces. If we only test that once I'm happy.

Reply:
I'd say that it's not the point of every single assertion we do, but we do want to test for the general format of lines. Just not for every assertion on the same line.